.github/actions/setup-python-env/action.yaml

```diff
@@ -67,7 +67,10 @@ runs:
 
   - name: Install Hatch
     shell: bash
-    run: pip3 install hatch
+    # Pin click to avoid bug in 8.3.0 - can remove when a fix is released
+    run: |
+      pip3 install "click<8.3.0"
```

Comment:
Do you have more context as to why this needs to be installed separately?

Reply:
It started blocking CI. There was a new version of click that was released today that broke our hatch install with python 3.12. Super random, took forever to debug.

Reply:
The bug was related to the `--filter` flag in hatch, which we aren't using, but it populated that param with a default value of a `Sentinel` object (whatever that is) that was not valid in JSON parsing.

Reply:
Here's a link to the CI failure if you're interested: https://github.com/dbt-labs/metricflow/actions/runs/17837131036/job/50717482593